src/App.vue

```diff
@@ -279,12 +287,31 @@ export default class App extends Mixins(BcolMixin, DateMixin, FilingTemplateMixi
           this.saveErrorDialog = true
       }
     })
+
+    this.$root.$on('name-request-invalid-error', async error => {
+      console.log('NR error during File and Pay =', error)
+      this.fileAndPayInvalidNameRequestDialog = true
+    })
+
+    this.$root.$on('name-request-retrieve-error', async () => {
+      console.log('Error while retrieving NR during File and Pay')
+      this.nameRequestInvalidErrorDialog = true
+    })
   }
 
   /** Called when component is destroyed. */
   private destroyed (): void {
     // stop listening for save error event
     this.$root.$off('save-error-event')
+    this.$root.$off('name-request-invalid-errort')
+    this.$root.$off('name-request-retrieve-error')
+  }
+
+  private goToManageBusinessDashboard () : void {
+    this.fileAndPayInvalidNameRequestDialog = false
+    const manageBusinessUrl = `${sessionStorage.getItem('AUTH_URL')}business`
+    window.onbeforeunload = null
```

Comment:
You should clear the 'haveChanges' flag instead.

Reply:
This is not for the custom popup. This is to disable the browsers default behavior

Reply:
What default behaviour?

Reply:
Nvm. I got it

Reply:
Done